static/app/components/tours/tourContext.tsx

```diff
@@ -74,22 +74,21 @@ type TourRegistry<T extends TourEnumType> = {
   [key in T]: TourStep<T> | null;
 };
 
-type TourCallbacks<T extends TourEnumType> = Partial<{
+type TourOptions<T extends TourEnumType> = Partial<{
   onEndTour: () => void;
   onStartTour: (stepId?: T) => void;
+  onStepChange: (stepId: T) => void;
+  requireAllStepsRegistered: boolean;
 }>;
 
 function tourReducer<T extends TourEnumType>(
   state: TourState<T>,
   action: TourAction<T>
 ): TourState<T> {
   const completeTourState = {...state, currentStepId: null, isCompleted: true};
+
   switch (action.type) {
     case 'START_TOUR': {
-      // If steps are not all registered, do nothing
-      if (!state.isRegistered) {
```

Comment:
Moved this to the `startTour()` function so that we can disable that requirement for the nav tour